onnxscript/ir/serde.py

```diff
@@ -1169,10 +1232,11 @@ def serialize_value_into(
         value_info_proto.name = from_.name
     if from_.metadata_props:
         _serialize_metadata_props_into(value_info_proto.metadata_props, from_.metadata_props)
-    if from_.shape is not None:
-        serialize_shape_into(value_info_proto.type, from_.shape)
     if from_.type is not None:
         serialize_type_into(value_info_proto.type, from_.type)
+    # Need to create the type _before_ writing the shape so that the shape can be written to the leaf type proto
```

Comment:
Should we show this dependency in code? Like putting from_.shape under from_.type is not None branch?

Reply:
Do you mean line 1238-1239?

Reply:
Yes

Reply:
I see what you are saying. Currently we just assume the type to be a type that is an unknown tensor dtype with a known shape, if type is None. I think that's ok?